### ISSUE TITLE
Bug: unzip_and_validated_gtfs - negsignal.SIGKILL

### ISSUE THREAD

**Issue by vevetron** [cal-itp/data-infra#3368]
**Describe the bug**
Several different airflow runs have died with this error:
`[2024-06-10, 04:19:14 UTC] {gtfs_csv_to_jsonl_hourly.py:153} INFO - Parsed gs://calitp-gtfs-schedule-unzipped-hourly/stop_times.txt/dt=2024-06-10/ts=2024-06-10T03:00:19.181898+00:00/base64_url=aHR0cHM6Ly9hcGkuNTExLm9yZy90cmFuc2l0L2RhdGFmZWVkcz9vcGVyYXRvcl9pZD1NQg==/stop_times.txt
[2024-06-10, 04:19:14 UTC] {gtfs_csv_to_jsonl_hourly.py:115} INFO - Processing gs://calitp-gtfs-schedule-unzipped-hourly/stop_times.txt/dt=2024-06-10/ts=2024-06-10T03:00:19.181898+00:00/base64_url=aHR0cHM6Ly9hcGkuNTExLm9yZy90cmFuc2l0L2RhdGFmZWVkcz9vcGVyYXRvcl9pZD1NVg==/stop_times.txt
[2024-06-10, 04:19:14 UTC] {storage.py:263} INFO - saving 15.5 kB to gs://calitp-gtfs-schedule-parsed-hourly/stop_times/dt=2024-06-10/ts=2024-06-10T03:00:19.181898+00:00/base64_url=aHR0cHM6Ly9hcGkuNTExLm9yZy90cmFuc2l0L2RhdGFmZWVkcz9vcGVyYXRvcl9pZD1NVg==/stop_times.jsonl.gz
[2024-06-10, 04:19:14 UTC] {gtfs_csv_to_jsonl_hourly.py:153} INFO - Parsed gs://calitp-gtfs-schedule-unzipped-hourly/stop_times.txt/dt=2024-06-10/ts=2024-06-10T03:00:19.181898+00:00/base64_url=aHR0cHM6Ly9hcGkuNTExLm9yZy90cmFuc2l0L2RhdGFmZWVkcz9vcGVyYXRvcl9pZD1NVg==/stop_times.txt
[2024-06-10, 04:19:14 UTC] {gtfs_csv_to_jsonl_hourly.py:115} INFO - Processing gs://calitp-gtfs-schedule-unzipped-hourly/stop_times.txt/dt=2024-06-10/ts=2024-06-10T03:00:19.181898+00:00/base64_url=aHR0cHM6Ly9hcGkuNTExLm9yZy90cmFuc2l0L2RhdGFmZWVkcz9vcGVyYXRvcl9pZD1QRQ==/stop_times.txt
[2024-06-10, 04:19:14 UTC] {storage.py:263} INFO - saving 51.4 kB to gs://calitp-gtfs-schedule-parsed-hourly/stop_times/dt=2024-06-10/ts=2024-06-10T03:00:19.181898+00:00/base64_url=aHR0cHM6Ly9hcGkuNTExLm9yZy90cmFuc2l0L2RhdGFmZWVkcz9vcGVyYXRvcl9pZD1QRQ==/stop_times.jsonl.gz
[2024-06-10, 04:19:14 UTC] {gtfs_csv_to_jsonl_hourly.py:153} INFO - Parsed gs://calitp-gtfs-schedule-unzipped-hourly/stop_times.txt/dt=2024-06-10/ts=2024-06-10T03:00:19.181898+00:00/base64_url=aHR0cHM6Ly9hcGkuNTExLm9yZy90cmFuc2l0L2RhdGFmZWVkcz9vcGVyYXRvcl9pZD1QRQ==/stop_times.txt
[2024-06-10, 04:19:14 UTC] {gtfs_csv_to_jsonl_hourly.py:115} INFO - Processing gs://calitp-gtfs-schedule-unzipped-hourly/stop_times.txt/dt=2024-06-10/ts=2024-06-10T03:00:19.181898+00:00/base64_url=aHR0cHM6Ly9hcGkuNTExLm9yZy90cmFuc2l0L2RhdGFmZWVkcz9vcGVyYXRvcl9pZD1SRw==/stop_times.txt
[2024-06-10, 04:20:31 UTC] {local_task_job.py:212} INFO - Task exited with return code Negsignal.SIGKILL
[2024-06-10, 04:20:31 UTC] {taskinstance.py:2599} INFO - 0 downstream tasks scheduled from follow-on schedule check
`

https://b2062ffca77d44a28b4e05f8f5bf4996-dot-us-west2.composer.googleusercontent.com/log?execution_date=2024-06-10T03%3A00%3A00%2B00%3A00&task_id=stop_times_txt&dag_id=unzip_and_validate_gtfs_schedule_hourly&map_index=-1

https://b2062ffca77d44a28b4e05f8f5bf4996-dot-us-west2.composer.googleusercontent.com/log?execution_date=2024-06-09T03%3A00%3A00%2B00%3A00&task_id=stop_times_txt&dag_id=unzip_and_validate_gtfs_schedule_hourly&map_index=-1

https://b2062ffca77d44a28b4e05f8f5bf4996-dot-us-west2.composer.googleusercontent.com/log?execution_date=2024-06-07T03%3A00%3A00%2B00%3A00&task_id=stop_times_txt&dag_id=unzip_and_validate_gtfs_schedule_hourly&map_index=-1

**To Reproduce**
every day during the 9pm run

**Expected behavior**
It finishes.

**Additional context**

**Comment by vevetron** [comment]
This is most likely caused by a [memory issue](https://cloud.google.com/composer/docs/composer-2/troubleshooting-dags#troubleshooting_sigterm_signals).

The final gtfs it chokes on is the bay area GTFS which is a lot larger.  

It might be possible to refactor the code so it handles the data in parts, but probably the best and easiest next step is to 
- increase the ram of the kubernetes image and see if the problem goes away.
- check the number of threads for this operation, reduce it if necessary

and just do it in production.

I'm not entirely sure how to do either of these.

**Comment by themightychris** [comment]
Ok now with #3348 fixed, we can now search Loki for log data containing `SIGKILL`: https://monitoring.calitp.org/explore?orgId=1&left=%7B%22datasource%22:%22eqM4SwA4k%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22airflow-jobs%5C%22%7D%20%7C%3D%20%60SIGKILL%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22eqM4SwA4k%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%22now-24h%22,%22to%22:%22now%22%7D%7D

Once a relevant line can be found, we can expand the line to identify the name of the Pod it ran on, and then on the [`Airflow and Jupyter resource consumption` dashboard](https://monitoring.calitp.org/d/ONcF4904k/airflow-and-jupyter-resource-consumption?orgId=1&refresh=30s) we can find that pod name and review its memory usage over time.

I'll need to come back to this after we have some data accumulated that shows crashes. I'll check back tomorrow

**Comment by vevetron** [comment]
I created a sigkill today (Aug 27th)
![image](https://github.com/user-attachments/assets/f54bacd4-cd0b-4e8a-8d76-97212d701a4a)

**Comment by vevetron** [comment]
The Thread: https://cal-itp.slack.com/archives/C01FNDG1ZPA/p1724808971547679

Based on @themightychris's investigation, he determined the relevant operator unzips use the GtfsGcsToJsonlOperatorHourly custom operator which uses Airflow directly.

He rightly argues "we’d need to convert those to run in pods, or increase the size of the Composer instance ($$$$)….but since there have been no crashes in prod since June… I feel like we should just close this and revisit if it ever starts happening again?"

We will attempt to re-run those failed runs in airflow (hit clear on the task that failed)

He successfully completed one of the tasks, i'll try to do the rest.

**Comment by vevetron** [comment]
<img width="606" alt="image" src="https://github.com/user-attachments/assets/42d55300-e16d-44d9-a68d-19747d20ec4a">

I ran the dag locally on my mac.  The download criteria includes 249 GTFS files to process.  Usually I allocate 8 gigs of ram but today I allocated 13.3.  From the image we see memory uses spikes to 12.61 when doing the `unzip_and_validate_gtfs_schedule_hourly` dag most likely when it hits the RG, regional feed.

I recommend we add instructions to airflow to allocate 12 gigs of ram to docker if you see a negsignal.SIGKILL error, or plan on running the whole pipeline.

2 jobs continue to fail in airflow.  Since we haven't seen a failure in many months, let's #wontfix for now otherwise solutions for to the future:
- Increase the memory on the docker-compose
- convert the operators to using pods
